src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -26,20 +25,116 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
     val openTelemetry: OpenTelemetryMetrics[IO]
 ) extends LazyLogging {
 
-  def createUser(user: SamUser, samRequestContext: SamRequestContext): IO[UserStatus] =
+  def createUser(possibleNewUser: SamUser, samRequestContext: SamRequestContext): IO[UserStatus] = {
     openTelemetry.time("api.v1.user.create.time", API_TIMING_DURATION_BUCKET) {
+      // Validate the values set on the possible new user, short circuit if there's a problem
+      val validationErrors = validateUser(possibleNewUser)
+      if (validationErrors.nonEmpty) {
+        return IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "invalid user", validationErrors.get)))
+      }
+
+      for {
+        newUser <- assertUserIsNotAlreadyRegistered(possibleNewUser, samRequestContext)
+        maybeWorkbenchSubject <- directoryDAO.loadSubjectFromEmail(newUser.email, samRequestContext)
+        registeredUser <- attemptToRegisterSubjectAsAUser(maybeWorkbenchSubject, newUser, samRequestContext)
+        registeredAndEnabledUser <- makeUserEnabled(registeredUser, samRequestContext)
+        _ <- addToAllUsersGroup(registeredAndEnabledUser.id, samRequestContext)
```

Comment:
I _think_ this call is unnecessary. When ` createUserInternal` is called, which happens when a user is invited or created, their proxy group is created and added to the all users group.